test/widgets/message_list_test.dart

```diff
@@ -294,6 +294,233 @@ void main() {
     });
   });
 
+  group('TypingStatusWidget', () {
+    final users = [eg.selfUser, eg.otherUser, eg.thirdUser, eg.fourthUser];
+    final finder = find.descendant(
+      of: find.byType(TypingStatusWidget),
+      matching: find.byType(Text)
+    );
+
+    Future<void> checkTyping(WidgetTester tester, TypingEvent event, {required String expected}) async {
+      await store.handleEvent(event);
+      await tester.pump();
+      check(tester.widget<Text>(finder)).data.equals(expected);
+    }
+
+    final dmMessage = eg.dmMessage(from: eg.selfUser, to: [eg.otherUser]);
+    final dmNarrow = DmNarrow.ofMessage(dmMessage, selfUserId: eg.selfUser.userId);
```

Comment:
Let's keep test data realistic — so if we're going to have eg.thirdUser and eg.fourthUser typing in this DM narrow, they should be members of it. (following up on https://github.com/zulip/zulip-flutter/pull/790#discussion_r1692326183 )